packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -115,8 +118,11 @@ export class Checkbox extends BaseComponent<ICheckboxProps, ICheckboxState> impl
         onBlur={ this._onBlur }
         aria-checked={ isChecked }
         aria-disabled={ disabled }
+        aria-label={ ariaLabel }
+        aria-labelledby={ ariaLabelledBy }
+        aria-describedby={ ariaDescribedBy }
       >
-        <label className={ this._classNames.label } htmlFor={ this._id } >
+        <label className={ this._classNames.label } htmlFor={ ariaLabelledBy || ariaLabel ? undefined : this._id } >
```

Comment:
Okay, looks like setting it to `undefined` would regress click behavior. If we set explicit `aria-label` the browser overrides the `htmlFor` binding and solves the confusion. Thanks for the review :)